### PR TITLE
Webdriver Switch Support with persistence

### PR DESCRIPTION
**Pull request by codemanat**
Added support to switch between Firefox and Chrome Webdrivers with session persistence

### CODE REVIEW

File: webwhatsapp.py [hunk 1]
```diff
@@ -1,21 +1,42 @@
 import time
+import os
+import sys
 
 from selenium import webdriver
 from selenium.webdriver.common.by import By
 from selenium.webdriver.support import expected_conditions as ec
 from selenium.webdriver.support.ui import WebDriverWait
+from selenium.webdriver.chrome.options import Options
+
```
Comment (moomoohk) [inline webwhatsapp.py]: Remove blank line

File: webwhatsapp.py [hunk 2]
```diff
@@ -1,21 +1,42 @@
...
-        self.wapi_functions = WapiJsWrapper(self._driver)
+    def __init__(self, browser='firefox', username="API"):
         self.username = username
+        if browser.lower() == 'firefox':
+            # from https://github.com/siddhant-varma/WhatsAPI/blob/master/webwhatsapi/__init__.py
+            self.path = os.path.dirname(sys.argv[0]) + 'firefox_cache' + '/' + self.username
+            if not os.path.exists(self.path):
+                os.makedirs(self.path)
+            self.profile = webdriver.FirefoxProfile(self.path)
+            #self._driver = webdriver.Firefox()
```
Comment (moomoohk) [inline webwhatsapp.py]: Remove comment

File: webwhatsapp.py [hunk 3]
```diff
@@ -1,21 +1,42 @@
...
+            self._driver = webdriver.Firefox(self.profile)
+        if browser.lower() == 'chrome':
+            self.chrome_options = Options()
```
Comment (moomoohk) [inline webwhatsapp.py]: Hide member: `self._chrome_options`

File: webwhatsapp.py [hunk 4]
```diff
@@ -1,21 +1,42 @@
...
+
 class WhatsAPIDriver(object):
-    def __init__(self, username="API"):
-        self._driver = webdriver.Firefox()
-        self.wapi_functions = WapiJsWrapper(self._driver)
+    def __init__(self, browser='firefox', username="API"):
         self.username = username
+        if browser.lower() == 'firefox':
+            # from https://github.com/siddhant-varma/WhatsAPI/blob/master/webwhatsapi/__init__.py
+            self.path = os.path.dirname(sys.argv[0]) + 'firefox_cache' + '/' + self.username
```
Comment (moomoohk) [inline webwhatsapp.py]: Use `os.path.join` to create paths not `+ '/' +`

File: webwhatsapp.py [hunk 5]
```diff
@@ -1,21 +1,42 @@
...
+            if not os.path.exists(self.path):
+                os.makedirs(self.path)
+            self.profile = webdriver.FirefoxProfile(self.path)
```
Comment (moomoohk) [inline webwhatsapp.py]: Hide and rename member: `self._firefox_profile`

File: webwhatsapp.py [hunk 6]
```diff
@@ -1,21 +1,42 @@
...
+            #self._driver = webdriver.Firefox()
+            self._driver = webdriver.Firefox(self.profile)
+        if browser.lower() == 'chrome':
+            self.chrome_options = Options()
+            self.chrome_options.add_argument("user-data-dir=" + os.path.dirname(sys.argv[0]) + 'chrome_cache' + '/' +
+                                             self.username )
+            self._driver = webdriver.Chrome(chrome_options=self.chrome_options)
+
+
```
Comment (moomoohk) [inline webwhatsapp.py]: Remove blank line

File: webwhatsapp.py [hunk 7]
```diff
@@ -1,21 +1,42 @@
...
+        self.wapi_functions = WapiJsWrapper(self._driver)
+
```
Comment (moomoohk) [inline webwhatsapp.py]: Remove blank line

File: webwhatsapp.py [hunk 8]
```diff
@@ -1,21 +1,42 @@
...
 from consts import Selectors, URL
 
 
+
+
 class WhatsAPIDriver(object):
-    def __init__(self, username="API"):
-        self._driver = webdriver.Firefox()
-        self.wapi_functions = WapiJsWrapper(self._driver)
+    def __init__(self, browser='firefox', username="API"):
```
Comment (moomoohk) [inline webwhatsapp.py:18]: Save browser type to a member: `self.browser_type = browser`

File: webwhatsapp.py [hunk 9]
```diff
@@ -1,21 +1,42 @@
...
         self.username = username
+        if browser.lower() == 'firefox':
+            # from https://github.com/siddhant-varma/WhatsAPI/blob/master/webwhatsapi/__init__.py
+            self.path = os.path.dirname(sys.argv[0]) + 'firefox_cache' + '/' + self.username
+            if not os.path.exists(self.path):
+                os.makedirs(self.path)
+            self.profile = webdriver.FirefoxProfile(self.path)
+            #self._driver = webdriver.Firefox()
+            self._driver = webdriver.Firefox(self.profile)
+        if browser.lower() == 'chrome':
```
Comment (moomoohk) [inline webwhatsapp.py]: Insert a line break before this `if` (better readability)

File: webwhatsapp.py [hunk 10]
```diff
@@ -1,21 +1,42 @@
...
+from selenium.webdriver.chrome.options import Options
+
 
 from message import Message, MessageGroup
 from chat import Chat
 from wapi_js_wrapper import WapiJsWrapper
 from consts import Selectors, URL
 
 
+
```
Comment (moomoohk) [inline webwhatsapp.py]: Remove both of these blank lines

**Review by moomoohk — CHANGES_REQUESTED** [review]
Nice little PR :)
Fix it up and I'll merge

File: webwhatsapp.py (hunk 8 above)
Comment (codemanat) [reply webwhatsapp.py:18]: Code changed as requested.

**Comment by codemanat** [comment]
Hello, i have changed the code as requested.

**Pull request merged**